java/src/main/java/com/useanvil/examples/client/GraphqlClient.java

```diff
@@ -0,0 +1,98 @@
+                .requestTimeout(Duration.ofSeconds(20))
+                .headersTimeout(Duration.ofSeconds(5))
+                .readTimeout(Duration.ofSeconds(5))
+                .autoAcceptEncoding(true)
+                .build();
+    }
+
+    public HttpRequest.Builder createRequestBuilder() throws RuntimeException {
+        return this.createRequestBuilder(Constants.GRAPHQL_ENDPOINT);
+    }
+
+    public HttpResponse<String> doRequest(Path queryFile) throws IOException, InterruptedException {
+        return this.doRequest(new String(Files.readAllBytes(queryFile)), null);
+    }
+
+    public HttpResponse<String> doRequest(Path queryFile, Map<String, Serializable> variables) throws IOException, InterruptedException {
+        String json = this._objectMapper.writeValueAsString(variables);
+        return this.doRequest(new String(Files.readAllBytes(queryFile)), json);
+    }
+
+    public HttpResponse<String> doRequest(Path queryFile, String variables) throws IOException, InterruptedException {
+        return this.doRequest(new String(Files.readAllBytes(queryFile)), variables);
+    }
+
+    public HttpResponse<String> doRequest(Path queryFile, CreateEtchPacket variables, Path[] files) throws IOException, InterruptedException {
+        // GraphQL requests are in the format `{ "query": "", "variables": "" }`
+        Map<String, Serializable> graphqlMap = new HashMap<>();
+        graphqlMap.put("query", new String(Files.readAllBytes(queryFile)));
+        graphqlMap.put("variables", variables);
+        String json = this._objectMapper.writeValueAsString(graphqlMap);
+
+        // Most of this is to handle multipart uploads.
+        // For normal files, a base64-encoded file in the payload should work
+        // fine, if the payload size is too large, you will have to use
+        // multipart as described in the spec here:
+        // https://github.com/jaydenseric/graphql-multipart-request-spec
+        //
+        // This will be handled automatically in a future Anvil Java library.
+        Map<String, String[]> map = new HashMap<>();
+        map.put("1", new String[]{"variables.files.1.file"});
+
+        // We're using Methanol's Multipart request builder here since it's
+        // easier to deal with for our purposes here.
+        // It's compatible with the built-in Java 11 HTTP client, so we can
+        // just pass that into `client.send()`.
+        MultipartBodyPublisher multipartBody = MultipartBodyPublisher.newBuilder()
+                .formPart("operations", HttpRequest.BodyPublishers.ofString(json))
+                .formPart("map", HttpRequest.BodyPublishers.ofString(this._objectMapper.writeValueAsString(map)))
+                .filePart("1", files[0])
+                .build();
+
+        HttpRequest request = this.createRequestBuilder()
+                .POST(multipartBody)
+                .build();
+
+        return this.client.send(request, HttpResponse.BodyHandlers.ofString());
+    }
```

Comment:
Multipart stuff here. Hardcoded to the only file we have, but should be easy to generalize in a library later.